### PR TITLE
Add Kerberos support for LDAP-Resolver

### PR DESCRIPTION
**Pull request by plettich**
Also add simple authentication support for users with GSSAPI.

Working on #770

### CODE REVIEW

File: privacyidea/static/components/config/views/config.policies.details.html [hunk 1]
```diff
@@ -403,6 +403,9 @@ <h3 translate>Action</h3>
                                             ng-bind-html="action.desc | highlight: action_filter">
                                             </p>
                                         </td>
+                                        <td>
+                                            <a href="{{ 'https://privacyidea.readthedocs.io/en/latest/' + selectedScope + '.html#' + action.name }}">help</a>
```
Comment (cornelinux) [inline privacyidea/static/components/config/views/config.policies.details.html:407]: We have these nice (i) icons.

    <span class="glyphicon glyphicon-info-sign"></span>

we can preceed the help link with that.

File: privacyidea/lib/resolvers/LDAPIdResolver.py [hunk 2]
```diff
@@ -329,7 +333,23 @@ def checkPass(self, uid, password):
...
-        if self.authtype == AUTHTYPE.NTLM:  # pragma: no cover
+        if self.authtype == AUTHTYPE.SASL_KERBEROS:
+            # we need to check credentials with kerberos differently since we
+            # can not use bind for every user
+            try:
+                import gssapi
+            except ImportError:
+                log.warning('Package "gssapi" missing. Can not authenticate users!')
+                return False
+            name = gssapi.Name(self.getUserInfo(uid).get('username'))
```
[the remark this answers is not in the document]
Comment (plettich) [reply privacyidea/lib/resolvers/LDAPIdResolver.py:344]: looks like this works with sAMAccountName and UPN (other login attributes we have to test).

File: privacyidea/static/components/config/views/config.policies.details.html (hunk 1 above)
Comment (plettich) [reply privacyidea/static/components/config/views/config.policies.details.html:407]: i think this is already in master... will check.

File: privacyidea/static/components/config/controllers/configControllers.js [hunk 3]
```diff
@@ -996,10 +985,7 @@ myApp.controller("LdapResolverController", ["$scope", "ConfigFactory", "$state",
         $scope.params.USERINFO = '{ "phone" : "telephoneNumber", "mobile" : "mobile", "email" : "mail", "surname" : "sn", "givenname" : "givenName" }';
         $scope.params.NOREFERRALS = true;
         $scope.params.EDITABLE = false;
-        $scope.params.SIZELIMIT = 500;
```
[the remark this answers is not in the document]
Comment (plettich) [reply privacyidea/static/components/config/controllers/configControllers.js:999]: I think the AD preset is only for configuring the LDAP attributes, not for configuring the (general) server settings.

File: privacyidea/lib/resolvers/LDAPIdResolver.py [hunk 4]
```diff
@@ -796,7 +818,7 @@ def loadConfig(self, config):
         self.object_classes = [cl.strip() for cl in config.get("OBJECT_CLASSES", "").split(",")]
         self.dn_template = config.get("DN_TEMPLATE", "")
         self.bindpw = config.get("BINDPW")
-        self.timeout = float(config.get("TIMEOUT", 5))
+        self.timeout = int(config.get("TIMEOUT", 5))
```
[the remark this answers is not in the document]
Comment (plettich) [reply privacyidea/lib/resolvers/LDAPIdResolver.py:821]: It looks like this is used for the (LDAP) Server timeout as well as for the Connections receive_timeout. One uses float, the other int... If we stick to int we should be fine.

File: privacyidea/lib/resolvers/LDAPIdResolver.py [hunk 5]
```diff
@@ -329,7 +333,23 @@ def checkPass(self, uid, password):
         -         false if password does not match
 
         """
-        if self.authtype == AUTHTYPE.NTLM:  # pragma: no cover
+        if self.authtype == AUTHTYPE.SASL_KERBEROS:
+            # we need to check credentials with kerberos differently since we
+            # can not use bind for every user
+            try:
+                import gssapi
```
[the remark this answers is not in the document]
Comment (plettich) [reply privacyidea/lib/resolvers/LDAPIdResolver.py]: Done in 8821f3bd

File: privacyidea/static/components/config/views/config.policies.details.html (hunk 1 above)
Comment (plettich) [reply privacyidea/static/components/config/views/config.policies.details.html]: Done in 8821f3bd

**Review by cornelinux — APPROVED** [review]

**Pull request merged**